lib/faraday/error.rb

```diff
@@ -83,6 +83,14 @@ def initialize(exc = 'timeout', response = nil)
     end
   end
 
+  # Raised by Faraday::Response::RaiseError in case of a nil status in response.
+  class NilStatusError < ServerError
+    def initialize(response = nil)
```

Comment:
I submitted a PR to your fork, this will need two params to handle the stack trace that's sent

Reply:
I saw that, just lifted what you had and popped it in here, easier than merging things into the fork